src/main/scala/symsim/concrete/ConcreteQTable.scala

```diff
@@ -0,0 +1,66 @@
+package symsim
+package concrete
+
+import cats.syntax.option.*
+import org.typelevel.paiges.Doc
+import symsim.QTable
+
+trait ConcreteQTable[State, ObservableState, Action]
+        extends QTable[State, ObservableState, Action, Double, Randomized]:
+  this: ConcreteExactRL[State, ObservableState, Action] =>
+
+  import agent.instances.*
+
+  type Scheduler[A] = Randomized[A]
+
+  def bestAction (q: Q) (s: State): Action =
+    val qs = q (agent.discretize (s)) map { _.swap }
+    qs (qs.keys.max)
+
+  def chooseAction (q: Q) (s: State): Scheduler[Action] =
+    for
+      explore <- Randomized.coin (this.epsilon)
+      action <- if explore
+      then Randomized.oneOf (allActions: _*)
```

Comment:
please indent 2 spaces more, and change `allActions: _*` to `allActions*` (new syntax)